### PR TITLE
playerattacktimer: plugin

### PR DESCRIPTION
**Pull request by dutta64**

### CODE REVIEW

**Comment by CJP10** [comment]
LGTM! I'll add some default animation id's in another PR.

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	tags = {"player", "attack", "tick", "timer", "delay"},
+	type = PluginType.UTILITY
+)
+public class PlayerAttackTimerPlugin extends Plugin
+{
+	private static final Map<Integer, Integer> animationTickMap = new HashMap<>();
+
+	static
+	{
+		animationTickMap.put(1658, 4); // Abyssal whip
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:68]: you would be better off making a PlayerAttackTimerTickMap class and putting them in there, that way if we add more we arent going to bloat the plugin.

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+			case HOPPING:
+				if (enabled)
+				{
+					shutDown();
+				}
+				break;
+		}
+	}
+
+	private void onGameTick(final GameTick event)
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:164]: @Subscribe

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	{
+		if (ticksUntilNextAnimation > 0)
+		{
+			--ticksUntilNextAnimation;
+		}
+	}
+
+	private void onAnimationChanged(final AnimationChanged event)
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:172]: @subscribe

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+	private void init()
+	{
+		enabled = true;
+
+		parseCustomAnimationConfig(config.customAnimations());
+
+		overlayManager.add(playerOverlay);
+
+		eventBus.subscribe(GameTick.class, this, this::onGameTick);
+		eventBus.subscribe(AnimationChanged.class, this, this::onAnimationChanged);
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:200]: can be removed due to adding @subscribe

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+		customAnimationTickMap.clear();
+
+		enabled = false;
+	}
+
+	@Subscribe
+	private void onConfigChanged(final ConfigChanged event)
+	{
+		if (!event.getGroup().equals("playerattacktimer"))
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:130]: doesnt account for mirror mode compatability, would need to reload plugin for it to take effect currently.

File: playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+			return;
+		}
+
+		init();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		eventBus.unregister(this);
```
Comment (xKylee) [inline playerattacktimer/src/main/java/net/runelite/client/plugins/playerattacktimer/PlayerAttackTimerPlugin.java:118]: can be removed due to adding @subscribe

**Comment by dutta64** [comment]
@xKylee changes made

**Pull request merged**